projects/graph/social/social.py

```diff
@@ -54,12 +57,25 @@ def populateGraph(self, numUsers, avgFriendships):
         
 
         # Create friendships
-        all_friendships = list(combinations(range(1, numUsers+1), avgFriendships))
+        # --- Following approach has O(n^2) runtime because of combinations, gets slow when n>1000 ----
+        all_friendships = list(combinations(range(1, numUsers+1), 2))
```

Comment:
in this case scenario of populating a graph a quadratic would usually be the average case, best and worst case time based. So in terms of optimization, your code base meets expectations